### ISSUE TITLE
Change Request: Rule Tester omit specific properties for duplicate test case detection

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#18173]
### ESLint version

8.57.0

### What problem do you want to solve?

The RuleTester currently stringifies all test case properties for each test case to check for duplicate test cases (see [here](https://github.com/eslint/eslint/blob/main/lib/rule-tester/rule-tester.js#L866)).
This means some duplicate test cases are not discovered.
Test cases with different `name`s:
```javascript
ruleTester.run('rule', rule, {
	valid: [
		{ code: 'code' },
		{ name: 'name', code: 'code' },
	],
	invalid: [],
}),
```

Test cases with a different error matcher:
```javascript
ruleTester.run('rule', rule, {
	valid: [],
	invalid: [	
		{ code: 'code', errors: 1 },
		{ code: 'code', errors: [{ messageId: 'id' }] },
	],
}
```

Not fixable test cases one specifying `output` and the other not:
```javascript
ruleTester.run('rule', rule, {
	valid: [],
	invalid: [	
		{ code: 'code', errors: 1, output: null },
		{ code: 'code', errors: 1 },
	],
}
```

### What do you think is the correct solution?

The used serialize package offers a [replacer](https://github.com/samn/json-stable-stringify#replacer) method which could exclude the properties `name`, `errors` and `output`.
The other Rule Tester specific property is `only` but I do not think that developers want to be warned about duplicate test cases when focusing on a test case.

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Ah, good catch! This seems like a bug to me. The goal of duplicate test detection is to make sure you're not running the rule on the same code using the same options multiple times, and we're clearly not doing that. Marking as accepted.

**Comment by bmish** [comment]
Makes sense to fix this.
